pages/orgs/[org]/domains/[domain]/index.js

```diff
@@ -108,39 +110,86 @@ function OrgDomainsList() {
   };
 
   const fetchDomainData = (page = 1) => {
-    fetch("/api/domains/" + domain + "?includeConcepts=true&page=" + page)
-      .then((response) => {
-        if (!response.ok) {
-          throw new Error(`HTTP error! Status: ${response.status}`);
-        }
-        return response.json();
-      })
-      .then((data) => {
-        if (data) {
-          setDomainData(data);
-          setSubDomainData(data.data?.subdomains);
-          let filteredConcepts = data?.data?.concepts?.filter(
-            (concept) => concept.type === "Concept"
-          );
-          if (data?.data?.subdomains && data?.data?.subdomains.length > 0) {
-            setSelectedSubdomain(data?.data?.subdomains[0]?.url);
-            fetchConcepts(data?.data?.subdomains[0]?.url);
+    // Check if the domain ID is `hpt-non-pharmaceuticals` or `hpt-pharmaceuticals`
+    let cascadeUrl = null;
+
+    if (domain === "hpt-non-pharmaceuticals") {
+      // Use cascade query URL for HPT-non-pharmaceuticals
+      cascadeUrl = `${API_BASE_URL}/orgs/MOH-KENYA/sources/PPB/concepts/10-00000-02/$cascade/?cascadeLevels=1&method=sourceToConcepts&view=hierarchy&reverse=true&includeRetired=false`;
+    } else if (domain === "hpt-pharmaceuticals") {
+      // Use cascade query URL for HPT-pharmaceuticals
+      cascadeUrl = `${API_BASE_URL}/orgs/MOH-KENYA/sources/PPB/concepts/10-00000-01/$cascade/?cascadeLevels=1&method=sourceToConcepts&view=hierarchy&reverse=true&includeRetired=false`;
+    }
+
+    if (cascadeUrl) {
+      fetch(cascadeUrl)
+        .then((response) => {
+          if (!response.ok) {
+            throw new Error(`HTTP error! Status: ${response.status}`);
+          }
+          return response.json();
+        })
+        .then((data) => {
+          if (data && data.entry) {
+            // Set subdomains from API response
+            const dynamicSubdomains = data.entry.entries.map((entry) => ({
+              id: entry.id,
+              display_name: entry.display_name,
+              url: entry.url,
+            }));
+
+            setSubDomainData(dynamicSubdomains);
+
+            if (dynamicSubdomains.length > 0) {
+              setSelectedSubdomain(dynamicSubdomains[0].url);
+              fetchConcepts(dynamicSubdomains[0].url);
+            }
           } else {
-            setConcepts(filteredConcepts);
-            setCurrentConcepts(filteredConcepts);
-            setIsLoadingConcepts(false);
```

Comment:
These seem removed from the new `if (cascadeUrl)` condition. We need to keep the `if` block uniform with the `else` as pertains these setters